pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AvgAggregationFunction.java

```diff
@@ -67,10 +67,11 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
     BlockValSet blockValSet = blockValSetMap.get(_expression);
     if (_nullHandlingEnabled) {
       RoaringBitmap nullBitmap = blockValSet.getNullBitmap();
-      if (nullBitmap != null && !nullBitmap.isEmpty()) {
-        aggregateNullHandlingEnabled(length, aggregationResultHolder, blockValSet, nullBitmap);
-        return;
+      if (nullBitmap == null) {
```

Comment:
We shouldn't need to change it for AVG because we are already storing object instead of double